nixos/modules/virtualisation/virtualbox-guest.nix

```diff
@@ -27,6 +49,33 @@ in
       type = types.bool;
       description = lib.mdDoc "Whether to enable x11 graphics";
     };
+
+    conditionVirtualization = mkOption {
+      default = "oracle";
+      type = types.str;
+      description = lib.mdDoc ''
+        The virtualized environment in which the guest additions should be started.
+        E.g., "oracle" or "kvm"
```

Comment:
Does it actually work with `kvm`?

Reply:
It should not. But on some devices virtualized environment can differ from "oracle". On one system I had to set "kvm".   I can remove 'or "kvm"' if you like.

Reply:
I'm also not sure why you would want to use this with KVM.

Reply:
I use a security focused, commercial system with Virtualbox as hypervisor on top of a custom Linux. Inside the VM 'systemd-detect-virt' returns: kvm. Maybe the revival of a very old systemd-bug? Anyway, the VBguest-service won't start with ConditionVirtualization="oracle". Being able to change the ConditionVirtualization-parameter would help me here.

off topic: I could not reproduce the detection error with a regular VB7 on Linux. Systemd detects "oracle" with all  paravirtualization interfaces in VB: kvm, hyper-v, minimal...

Reply:
> It should not. But on some devices virtualized environment can differ from "oracle". On one system I had to set "kvm". I can remove 'or "kvm"' if you like.

That's a systemd bug then. I say we get rid of the option, always set it to `oracle`, people with weird systemd bugs can override it at systemd level (and report the bug to systemd).